api/server/router/local/image.go

```diff
@@ -137,6 +138,12 @@ func (s *router) postImagesCreate(ctx context.Context, w http.ResponseWriter, r
 				err = s.daemon.PullImage(ref, metaHeaders, authConfig, output)
 			}
 		}
+		// Check the error from pulling an image to make sure the request
+		// was authorized. Modify the status if the request was
+		// unauthorized to respond with 401 rather than 500.
+		if err != nil && isAuthorizedError(err) {
```

Comment:
When I pull a private image from hub, this `err` is an `*errors.errorString` with the text `Error: image <image> not found`